test_cases/solidity/insufficient_gas_griefing/relayer/relayer.sol

```diff
@@ -0,0 +1,27 @@
+/*
+ * @source: https://consensys.github.io/smart-contract-best-practices/known_attacks/#insufficient-gas-griefing
+ * @author: ConsenSys Diligence
+ * Modified by Kaden Zipfel
+ */
+
+pragma solidity ^0.4.21;
+
+contract Relayer {
+    mapping (bytes => bool) executed;
```

Comment:
The problem with this approach, is that you never call the same data twice.  This would be problematic if for example the Executor was an ERC20 token, and you wanted to pay the 10 tokens to the same address more than once.

Reply:
Take a look at gnosis multisig, or https://github.com/christianlundkvist/simple-multisig to see how they handle it.